ts/features/payments/onboarding/screens/PaymentsOnboardingFeedbackScreen.tsx

```diff
@@ -68,6 +72,11 @@ const PaymentsOnboardingFeedbackScreen = () => {
   const rptIdToResume = useIOSelector(selectPaymentOnboardingRptIdToResume);
   const { startPaymentFlow } = usePagoPaPayment();
   const { bottomSheet, present } = usePaymentOnboardingAuthErrorBottomSheet();
+  const supportModal = usePaymentFailureSupportModal({
+    outcome,
+    isOnboarding: true
+  });
+  const paymentMethodSelectedRef = React.useRef<string | undefined>();
```

Comment:
Why do we use `useRef` here and not `useState`?

Reply:
I used `useRef` here instead of `useState` to avoid unnecessary re-rendering once the selected payment method is set. The payment method is reset after the first render, so it’s crucial to save it specifically during the `onFirstRender` lifecycle. By using `useRef`, we ensure that the value persists without triggering additional renders.

Reply:
Make sense!